source/agora/node/Validator.d

```diff
@@ -423,4 +430,54 @@ public class Validator : FullNode, API
         if (this.config.admin.enabled)
             this.admin_interface.stop();
     }
+
+    /***************************************************************************
+
+        Check if the current enrollment is about to expire or validator is not
+        enrolled. Enroll when necessary.
+
+        Params:
+            block_height = Current block height
+
+    ***************************************************************************/
+
+    private void checkAndEnroll (Height block_height) @safe
+    {
+        Hash enroll_key = this.enroll_man.getEnrolledUTXO(this.utxo_set.getUTXOFinder());
+
+        if (enroll_key == Hash.init && (enroll_key = this.getFrozenUTXO()) == Hash.init)
+            return; // Not enrolled and no frozen UTXO
+
+        const enrolled = this.enroll_man.getEnrolledHeight(enroll_key);
+
+        // This validators enrollment will expire next cycle or not enrolled at all
+        if (enrolled == ulong.max || block_height + 1 >= enrolled + this.params.ValidatorCycle)
+        {
+            log.trace("Enrolling at height {} for {} cycles with {}", block_height,
+                                            this.params.ValidatorCycle , enroll_key);
```

Comment:
```suggestion
            log.trace("Sending Enrollment at height {} for {} cycles with {}", block_height,
                                            this.params.ValidatorCycle , enroll_key);
```

It's important to have the distinction of "sending stuff" vs things actually being applied.